### PR TITLE
Project and sprint API functionality

### PR DESCRIPTION
**Pull request by hardins01**
Holyyyy, this is a big one. Here's a list of the stuff I did:

- Components I created:
  - `ProjectCardComponent` to separate out the functionality for cards on the projects-page
  - `ColorInplaceComponent` for editing the color of a project (useful for labels too)
  - `FundsInplaceComponent` for editing the funds of a project 
- Services I created:
  - `ProjectService` to work with projects in the backend
    - Creating, updating, deleting, getting projects
    - Had to update many components to use the new projectservice (new ones, description inplace, date inplace, title inplace, projects page, project page, nav menu, gantt page, create project modal)
  - `SprintService` to work with sprints in the backend
    - Creating, updating, deleting, getting sprints
    - Had to update many components to use the new sprintservice (description inplace, date inplace, title inplace, create sprint modal, sprint dropdown)
- Everything else:
  - Completed `ProjectData`
    - Added the rest of the attributes it needed to match backend
    - Helper function for converting between backend and frontend implementations
  - Completed `SprintData`
    - Added the rest of the attributes it needed to match backend
    - Helper function for converting between backend and frontend implementations
  - Several updates to project page
    - Because projects are bound to be pretty large, I added a loading spinner and animation while the page loads
      - It already shows the spinner a decent amount with small projects, which is way better than a blankish page
    - Added description inplace
    - Added the new color inplace and funds inplace
    - Added labels for better usability
  - A couple updates to sprint dropdown
    - Added description-inplace for updating the sprint's `goal`
    - Added labels for better usability
  - Improved create project modal
    - Finished adding all the project's attributes
    - Added validation before input is allowed
  - Improved create sprint modal
    - Finished adding all the sprint's attributes
    - Added validation before input is allowed
  - Now getting the user's name for the welcome message on the projects page
  - Got rid of extra code on the home page we aren't using anymore
  - Added border to description inplace to make it more cohesive and hopefully help people to see that it's clickable
  - Added a defaultText value to description inplace when the description is an empty string

### CODE REVIEW

File: ClientApp/src/app/components/inplaces/color-inplace/color-inplace.component.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import { ProjectService } from 'src/app/services/project/project.service';
+import { ProjectData } from 'src/types/project';
+
+@Component({
+  selector: 'color-inplace',
+  templateUrl: './color-inplace.component.html',
+  styleUrls: ['./color-inplace.component.scss'],
+  providers: [MessageService],
+})
+export class ColorInplaceComponent implements OnInit {
```
Comment (NathanJesudason) [inline ClientApp/src/app/components/inplaces/color-inplace/color-inplace.component.ts:18]: Is there a place where we would use this for labels as well?

File: ClientApp/src/app/components/inplaces/description-inplace/description-inplace.component.html [hunk 2]
```diff
@@ -1,8 +1,11 @@
 <p-toast></p-toast>
 
-<div id="displayedText" [innerHTML]="text" *ngIf="!selected" (click)="select()">
```
Comment (NathanJesudason) [inline ClientApp/src/app/components/inplaces/description-inplace/description-inplace.component.html:3]: Was this from a linter?

File: ClientApp/src/app/animations/animations.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      animate('300ms 0ms ease-out', style({ height: '*' })),
+    ]),
+  ]),
+  transition(':leave', [
+    group([
+      animate('300ms 0ms ease-in', style({ opacity: 0 })),
+      animate('300ms 0ms ease-in', style({ height: 0 })),
+    ]),
+  ]),
+]);
```
Comment (stef-timmermans) [inline ClientApp/src/app/animations/animations.ts:26]: Need line at EOF

File: ClientApp/src/app/components/modals/create-project-modal/create-project-modal.component.html [hunk 4]
```diff
@@ -22,25 +22,57 @@
...
+          <span class="ql-formats">
+            <button type="button" class="ql-list" value="ordered" aria-label="Ordered List"></button>
+            <button type="button" class="ql-list" value="bullet" aria-label="Bullet List"></button>
+          </span>
+        </ng-template>
+      </p-editor>
 
-      <p>Need to add more inputs in here when we're ready</p>
-    </div>
+      <p-colorPicker
```
Comment (NathanJesudason) [inline ClientApp/src/app/components/modals/create-project-modal/create-project-modal.component.html:47]: Do we want a title over the color picker like with the other inputs?

File: ClientApp/src/app/components/pages/project-page/project-page.component.ts [hunk 5]
```diff
@@ -27,8 +28,11 @@ import { SprintData } from 'src/types/sprint';
   templateUrl: './project-page.component.html',
   styleUrls: ['./project-page.component.scss'],
   providers: [MessageService],
+  animations: [BasicFadeAmination],
 })
 export class ProjectPageComponent {
+  pageLoading: boolean = true;
```
Comment (NathanJesudason) [inline ClientApp/src/app/components/pages/project-page/project-page.component.ts:34]: This is nice, I'll have to add this for the task page(s) too

File: ClientApp/src/app/services/project/project.service.spec.ts [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+import { ProjectService } from './project.service';
+
+describe('ProjectService', () => {
+  let projectServiceHttpSpy: jasmine.SpyObj<HttpClient>;
+  let sprintServiceHttpSpy: jasmine.SpyObj<HttpClient>;
+  let projectService: ProjectService;
+
+  beforeEach(() => {
+    projectServiceHttpSpy = jasmine.createSpyObj('HttpClient', ['get']);
```
Comment (NathanJesudason) [inline ClientApp/src/app/services/project/project.service.spec.ts:19]: This is nice, I should also add this to the task service

**Review by NathanJesudason — APPROVED** [review]
Fantastic Work!

**Review by yevgeniychaplygin3 — APPROVED** [review]
Looks Good

**Pull request merged**